cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -19,26 +19,21 @@
 #include <cstring>
 #include <limits>
 
-#include "arrow/array/array_base.h"
 #include "arrow/array/array_binary.h"
 #include "arrow/array/array_dict.h"
 #include "arrow/array/array_nested.h"
 #include "arrow/array/builder_primitive.h"
 #include "arrow/array/concatenate.h"
 #include "arrow/buffer_builder.h"
-#include "arrow/chunked_array.h"
 #include "arrow/compute/api_vector.h"
 #include "arrow/compute/kernels/common.h"
 #include "arrow/compute/kernels/util_internal.h"
 #include "arrow/extension_type.h"
-#include "arrow/record_batch.h"
 #include "arrow/result.h"
-#include "arrow/table.h"
```

Comment:
I'm surprised those includes are not needed anymore. `vector_selection.cc` certainly processes tables, record batches and chunked arrays...

Reply:
Well, as a previous suggestion I move the includes to util_internal, so in the future, we won't need to include RecordBatch/Table/ChukedArray files, just include util_internal.

Reply:
I don't see where such a suggestion was made. In general, we want to minimize the number of transitive includes in a `.h`, to try to control compile times.